modules/c/examples/code_snippets/main.cpp

```diff
@@ -1067,17 +1064,46 @@ static void create_array_index_nested() {
     CBLCollection *collection = CBLDatabase_DefaultCollection(kDatabase, NULL);
 
     // tag::array-index-nested[]
-    CBLArrayIndexConfiguration config = {
-        kCBLN1QLLanguage,
-        FLSTR("contacts[].phones"),
-        FLSTR("type")
-    };
+    CBLArrayIndexConfiguration config{};
+    config.expressionLanguage = kCBLN1QLLanguage;
+    config.path = FLSTR("contacts[].phones");
+    config.expressions = FLSTR("type");
 
     CBLError err{};
     CBLCollection_CreateArrayIndex(collection, FLSTR("myindex"), config, &err);
     // end::array-index-nested[]
 }
 
+static void create_partial_value_index() {
+    CBLDatabase* database = kDatabase;
+    CBLCollection* collection = CBLDatabase_DefaultCollection(kDatabase, NULL);
+
+    // tag::partial-value-index[]
+    CBLValueIndexConfiguration config {};
+    config.expressionLanguage = kCBLN1QLLanguage;
+    config.expressions = FLSTR("city");
+    config.where = FLSTR("type = \"hotel\"");
+
+    CBLError err{};
+    CBLCollection_CreateValueIndex(collection, FLSTR("HotelCityIndex"), config, &err);
+    // end::partial-value-index[]
+}
+
+static void create_partial_full_text_index() {
+    CBLDatabase* database = kDatabase;
+    CBLCollection* collection = CBLDatabase_DefaultCollection(kDatabase, NULL);
+
+    // tag::partial-full-text-index[]
+    CBLFullTextIndexConfiguration config{};
+    config.expressionLanguage = kCBLN1QLLanguage;
+    config.expressions = FLSTR("content");
+    config.where = FLSTR("year > 1999");
```

Comment:
Made a comment in the ticket related to this - also see iOS PR. It doesn't make much of a difference, but would prefer if we all can have the same (no matter which one). Otherwise, LGTM.